diff -Naurp emulationstation-76c1538/es-app/src/guis/GuiMenu.cpp emulationstation-76c1538-patch/es-app/src/guis/GuiMenu.cpp
--- emulationstation-76c1538/es-app/src/guis/GuiMenu.cpp	2016-07-04 13:37:34.000000000 +0300
+++ emulationstation-76c1538-patch/es-app/src/guis/GuiMenu.cpp	2016-07-04 13:49:09.000000000 +0300
@@ -209,6 +209,16 @@ GuiMenu::GuiMenu(Window* window) : GuiCo
 			mWindow->pushGui(new GuiDetectDevice(mWindow, false, nullptr));
 	});
 
+	addEntry("RETROARCH", 0x777777FF, true, 
+		[this] {
+						remove("/var/lock/start.games");
+						runSystemCommand("touch /var/lock/start.retro");
+						runSystemCommand("systemctl start retroarch.service");
+						SDL_Event ev;
+						ev.type = SDL_QUIT;
+						SDL_PushEvent(&ev);
+	});
+
 	addEntry("REBOOT", 0x777777FF, true, 
 		[this] {
 						SDL_Event ev;
